### PR TITLE
Flip many specials to globally unique to prevent duplicate NPCs.

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
Content "Make most locations with unique NPCs globally unique"

#### Purpose of change
Core-game version of https://github.com/CleverRaven/Cataclysm-DDA/pull/63627
Lots of good detail and rationales there.
Avoids immersion-breaking issues around having clones of most NPCs wandering around in the game world.

#### Describe the solution
Mark the associated overmap specials with the GLOBALLY_UNIQUE flag to prevent them spawning a second time.
Also increase search radius for missions that place these specials to avoid failures.

#### Describe alternatives you've considered
#63627 adds this as a mod, but that is going to fragment the player base and make it even harder to resolve bugs that are likely to arise from this kind of change.

#### Testing
Honestly #63627 did most of the due diligence here, I just did some spot checking.

### CODE REVIEW

**Comment by Maleclypse** [comment]
Might need to change some tests?
```
overmap_terrain_coverage
([slow] ~starting_items)=> -------------------------------------------------------------------------------
([slow] ~starting_items)=> ../tests/overmap_test.cpp:265
([slow] ~starting_items)=> ...............................................................................
([slow] ~starting_items)=> 
([slow] ~starting_items)=> ../tests/overmap_test.cpp:369: FAILED:
([slow] ~starting_items)=>   CHECK( num_missing == 0 )
([slow] ~starting_items)=> with expansion:
Error: ([slow] ~starting_items)=>   10 == 0
([slow] ~starting_items)=> with messages:
([slow] ~starting_items)=>   num_missing := 10
([slow] ~starting_items)=>   missing := { int_id( "chemical_lab_ocu" ), int_id( "chemical_lab_roof_ocu" ),
([slow] ~starting_items)=>   int_id( "smallscrapyard_ocu" ), int_id( "lumbermill_0_0_ocu" ), int_id(
([slow] ~starting_items)=>   "lumbermill_0_1_ocu" ), int_id( "lumbermill_1_0_ocu" ), int_id(
([slow] ~starting_items)=>   "lumbermill_1_1_ocu" ), int_id( "homelesscamp_npc" ), int_id(
([slow] ~starting_items)=>   "lodge_ground_glooscap1" ), int_id( "lodge_ground_glooscap2" ) }
([slow] ~starting_items)=>   To resolve errors about missing terrains you can either give the terrain the
([slow] ~starting_items)=>   SHOULD_NOT_SPAWN flag (intended for terrains that should never spawn, for
([slow] ~starting_items)=>   example test terrains or work in progress), or tweak the constraints so that
([slow] ~starting_items)=>   the terrain can spawn more reliably, or add them to the whitelist above in
([slow] ~starting_items)=>   this function (inteded for terrains that sometimes spawn, but cannot be
([slow] ~starting_items)=>   expected to spawn reliably enough for this test)
([slow] ~starting_items)=> 
([slow] ~starting_items)=> Log messages during failed test:
([slow] ~starting_items)=> 12:00:00AM: Unable to place all configured specials, some missions may fail to initialize.
([slow] ~starting_items)=> 12:00:00AM: Unable to place all configured specials, some missions may fail to initialize.
([slow] ~starting_items)=> 12:00:00AM: Unable to place all configured specials, some missions may fail to initialize.
([slow] ~starting_items)=> 12:00:00AM: Unable to place all configured specials, some missions may fail to initialize.
([slow] ~starting_items)=> 290.376 s: overmap_terrain_coverage
```

**Comment by kevingranade** [comment]
Clearly I missed something,  I'll look up what the other globally_unique prs did.

**Comment by Perset** [comment]
I will say in promotion of my very, very simple mod that I did a thorough playthrough of the game and had no issues. That said, I also raised some issues about how the mod works on Discord (I'm 0--) and in the thread itself, so I hope those things are tolerable. I remember in particular that I thought the Exodii signposts might cause search failures because they spawn infinitely searching for a now globally unique tile, which I was completely fine with. Anyway, I'm honored.

**Comment by Procyonae** [comment]
GLOBALLY_UNIQUE and UNIQUE both make "occurrences": [ x, y ] an x in y chance to try to spawn the location as opposed to between x and y attempts without it so it just needs upping to sensible numbers or [ 99, 100 ] to almost always attempt it once.

File: data/json/overmap/overmap_special/specials.json [hunk 1]
```diff
@@ -506,7 +506,7 @@
     "city_distance": [ 30, -1 ],
     "city_sizes": [ 0, 12 ],
     "occurrences": [ 0, 1 ],
```
Comment (Procyonae) [inline data/json/overmap/overmap_special/specials.json:508]:
```suggestion
    "occurrences": [ 30, 100 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 2]
```diff
@@ -7797,7 +7797,7 @@
     "city_distance": [ 5, 10 ],
     "city_sizes": [ 4, -1 ],
     "occurrences": [ 0, 1 ],
```
Comment (Procyonae) [inline data/json/overmap/overmap_special/specials.json:7799]:
```suggestion
    "occurrences": [ 30, 100 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 3]
```diff
@@ -7808,7 +7808,7 @@
     "city_distance": [ 5, 40 ],
     "city_sizes": [ 1, -1 ],
     "occurrences": [ 0, 1 ],
```
Comment (Procyonae) [inline data/json/overmap/overmap_special/specials.json:7810]:
```suggestion
    "occurrences": [ 30, 100 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 4]
```diff
@@ -7839,7 +7839,7 @@
     "city_distance": [ 5, 60 ],
     "city_sizes": [ 4, -1 ],
     "occurrences": [ 0, 2 ],
```
Comment (Procyonae) [inline data/json/overmap/overmap_special/specials.json:7841]:
```suggestion
    "occurrences": [ 60, 100 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 5]
```diff
@@ -7849,7 +7849,7 @@
     "city_distance": [ 3, 10 ],
     "city_sizes": [ 8, -1 ],
     "occurrences": [ 0, 1 ],
```
Comment (Procyonae) [inline data/json/overmap/overmap_special/specials.json:7851]:
```suggestion
    "occurrences": [ 30, 100 ],
```

**Review by Procyonae — COMMENTED** [review]
Just some suggested values based on their current occurrences ^^

**Comment by Procyonae** [comment]
Rather than just continually adding rare stuff to overmap_test would it not make more sense for the test itself to check how common it should be from the numbers and either ignore it if it's rare or test it in an area scaled to accommodate for it's rarity rather than the standard distance? Then the only things that would need to be manually added are ones with awkward spawn conditions where it can't judge the rarity from occurances + UNIQUE/GLOBALLY_UNIQUE.

**Comment by I-am-Erk** [comment]
I still wouldn't mind some more movement on making npcs less easy to accidentally kill first. #62349 is my personal request for how to resolve this problem. Then the mod can be to make story npcs unkillable using Downed status, rather than using a mod to invoke spawn behaviour that should be a default.

**Pull request closed without merging**